pastastore/extensions/hpd.py

```diff
@@ -317,14 +311,71 @@ def download_knmi_meteo(
             kind=kind,
             data_column=meteo_var,
             unit_multiplier=unit_multiplier,
+            update=False,
         )
 
+    def update_knmi_meteo(
+        self,
+        names: Optional[List[str]] = None,
+        tmin: TimeType = None,
+        tmax: TimeType = None,
+    ):
+        """Update meteorological data from KNMI in PastaStore.
+
+        Parameters
+        ----------
+        names : list of str, optional
+            list of names of observations to update, by default None
+        tmin : TimeType, optional
+            start time, by default None, which uses current last observation timestamp
+            as tmin
+        tmax : TimeType, optional
+            end time, by default None, which defaults to today
+        """
+        if names is None:
+            names = self._store.stresses.loc[
+                self._store.stresses["source"] == "KNMI"
+            ].index.tolist()
+
+        tmintmax = self._store.get_tmin_tmax("stresses", names=names)
+
+        for name in tqdm(names, desc="Updating KNMI meteo stresses"):
+            stn = self._store.stresses.loc[name, "station"]
+            meteo_var = self._store.stresses.loc[name, "meteo_var"]
+            unit = self._store.stresses.loc[name, "unit"]
+            kind = self._store.stresses.loc[name, "kind"]
+
+            if unit == "mm":
+                unit_multiplier = 1e3
+            else:
+                unit_multiplier = 1.0
+
+            if tmin is None:
+                tmin = tmintmax.loc[name, "tmax"]
+
+            knmi = hpd.read_knmi(
+                stns=[stn],
+                meteo_vars=[meteo_var],
+                starts=tmin,
+                ends=tmax,
+            )
+
+            self.add_observation(
+                "stresses",
+                knmi["obs"].iloc[0],
+                name=name,
+                kind=kind,
+                data_column=meteo_var,
+                unit_multiplier=unit_multiplier,
+                update=True,
+            )
+
     def download_bro_gmw(
         self,
-        extent=None,
-        tmin=None,
-        tmax=None,
-        update=False,
+        extent: Optional[List[float | int]] = None,
```

Comment:
I believe in type hinting an int is also a float 🤫 
Sou `Optional[List[float]]` would also do.

Reply:
https://peps.python.org/pep-0484/#the-numeric-tower